config/samples/policies/check-cluster-remote-access.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: check-cluster-remote-access
+  annotations:
+    policies.kyverno.io/title: Check Cluster Remote Access
+    policies.kyverno.io/category: EKS Best Practices
+    policies.kyverno.io/severity: medium
+    policies.kyverno.io/subject: Cluster
+    policies.kyverno.io/description: >-
+      Cluster remote access should be disabled.
+spec:
+  validationFailureAction: audit
+  background: true
+  rules:
+    - name: check-cluster-remote-access
+      match:
+        any:
+        - resources:
+            kinds:
+              - AWSAdapterConfig
+      validate:
+        message: "Remote access should be disabled. The field status.eksCluster.compute.nodeGroups must not define remoteAccessConfig."
+        pattern:
+          status:
+            eksCluster:
+              compute:
+                nodeGroups:
+                  - X(remoteAccessConfig): "null"
```

Comment:
Currently `remoteAccessConfig` is populated as a non-nil empty struct in its absence, with the same being the case for `launchTemplate`. This causes the policy to fail even when it should pass.

Let me know if `nil` is a better fit for such cases for writing policies, and I'll raise a PR to fix every such instance.

<img width="551" alt="image" src="https://user-images.githubusercontent.com/37282098/204931431-45c8f3cc-064f-490e-a74e-b08bbe87e163.png">

Reply:
We can also rewrite the policy if needed. Personally, I don't like empty structs as lazy placeholders for fields which should have values; I'd rather seem the omitted than included with an empty set.

Reply:
Merged the other PR where remoteAccessConfig and launchTemplate are set to Nil instead of empty struct.